appveyor/Convert-FileUrisToGitHubUris.ps1

```diff
@@ -50,20 +53,24 @@ function Get-CaseSensitivePath($pathName){
 $sarifLog = (Get-Content $logFilePath) -join "`n" | ConvertFrom-Json
 
 for ($i = 0; $i -lt $sarifLog.runs.Count; $i++){
-  $sarifLog.runs[$i].files.PSObject.Properties | foreach-object{
-    $key = $_.Name
-    $content = $_.Value
-    if($content.uri){
-        $content.uri = Rebase-Uri $content.uri
-    }
-
-    if($content.parentKey){
-      $content.parentKey = Rebase-Uri $content.parentKey
+  if($sarifLog.runs[$i].files){
+    $sarifLog.runs[$i].files.PSObject.Properties | foreach-object{
+      $key = $_.Name
+      $content = $_.Value
+      if($key -and $content){
+        if($content.uri){
+            $content.uri = Rebase-Uri $content.uri
+        }
+
+        if($content.parentKey){
+          $content.parentKey = Rebase-Uri $content.parentKey
+        }
+
+        $sarifLog.runs.files.PSObject.Properties.Remove($_.Name)
+        $rewrite = Rebase-Uri $key
```

Comment:
1. Variable names are generally nouns; "rewrite" is a verb.
2. You chose the name `Rebase-Uri` for your function (which is a good name), so in any case the verb should be "rebase" rather than "rewrite".

So... how about `$rebasedUri`?